apps/spruce/src/constants/routes.ts

```diff
@@ -287,8 +288,16 @@ export const getDistroSettingsRoute = (
 export const getCommitsRoute = (projectIdentifier: string = "") =>
   `${paths.commits}/${encodeURIComponent(projectIdentifier)}`;
 
-export const getWaterfallRoute = (projectIdentifier: string = "") =>
-  `${paths.project}/${encodeURIComponent(projectIdentifier)}${paths.waterfall}`;
+export const getWaterfallRoute = (
+  projectIdentifier?: string,
```

Comment:
Is there any reason not to require `projectIdentifier`?

Reply:
It avoids having to cast projectIdentifier/blocking the render when it comes from useParams or Apollo. Also `getWaterfallRoute` originally allowed an optional value so I preserved that functionality.